Dockerfile-probe

```diff
@@ -11,16 +10,14 @@ RUN ["go", "mod", "download"]
 
 # Build the probe binary
 ARG PROBE_VERSION
-RUN ["env", "CGO_ENABLED=0", "GOOS=linux", "go", "build", "-ldflags=-w -s -extldflags '-static' -X 'github.com/grafana/nethax/pkg/kubernetes.ProbeImageVersion=${PROBE_VERSION}'", "-a", "-o", "/probe", "./cmd/probe"]
```

Comment:
I had to change this to the non-exec syntax because Dockerfiles do not support escaping of quotes inside a RUN:
https://docs.docker.com/reference/dockerfile/#escape

I would gladly accept a fix for this, but I tried for >2h to get this working with the previous syntax and could not get it.